pkg/storaged/client.js

```diff
@@ -837,6 +906,125 @@ function vdo_overlay() {
 
 client.vdo_overlay = vdo_overlay();
 
+/* Stratis */
+
+client.stratis_start = () => {
+    var stratis = cockpit.dbus("org.storage.stratis2", { superuser: "try" });
+    client.stratis_manager = stratis.proxy("org.storage.stratis2.Manager.r1",
+                                           "/org/storage/stratis2");
+
+    client.stratis_pools = { };
+    client.stratis_blockdevs = { };
+    client.stratis_filesystems = { };
+    client.stratis_manager.data.LockedPoolsWithDevs = [];
+    client.stratis_manager.data.KeyList = [];
+
+    return client.stratis_manager.wait()
+            .then(() => {
+                console.log("STRATIS", client.stratis_manager.Version);
+
+                if (utils.compare_versions(client.stratis_manager.Version, "2.4") < 0)
+                    return Promise.reject(new Error("stratisd too old, need at least version 2.4"));
+
+                client.features.stratis = true;
+                client.stratis_pools = client.stratis_manager.client.proxies("org.storage.stratis2.pool.r1",
+                                                                             "/org/storage/stratis2",
+                                                                             { watch: false });
+                client.stratis_blockdevs = client.stratis_manager.client.proxies("org.storage.stratis2.blockdev.r2",
+                                                                                 "/org/storage/stratis2",
+                                                                                 { watch: false });
+                client.stratis_filesystems = client.stratis_manager.client.proxies("org.storage.stratis2.filesystem",
+                                                                                   "/org/storage/stratis2",
+                                                                                   { watch: false });
+
+                return stratis.watch({ path_namespace: "/org/storage/stratis2" }).then(() => {
+                    client.stratis_manager.client.addEventListener('notify', (event, data) => {
+                        console.log("NOTIFY", JSON.stringify(data));
+                        update_indices();
```

Comment:
The trio of update_indices, find_warnings, dispatchEvent should be factored out into client.update()

Reply:
Done.